src/ui/public/visualize/visualize_legend.js

```diff
@@ -66,6 +66,18 @@ uiModules.get('kibana')
         return filters.length;
       };
 
+      $scope.showLabelCount = legendData => {
+        if (!$scope.vis.params.addLegendCount) return false;
+
+        const isBarChart = $scope.vis.type.name === 'histogram';
+        const isPieChart = $scope.vis.type.name === 'pie';
+        return  (isBarChart || isPieChart) && $scope.getLabelCount(legendData);
```

Comment:
Hmmm... this type of check for specific chart types in this module is a bit of a code smell to me. I think a better check would be something like:

```js
const isLegendCountSupported = !!$scope.vis.type.isLegendCountSupported;
return isLegendCountSupported && $scope.getLegendCount(legendData);
```

That way you can define `isLegendCountSupported` to be `true` in the code that is specific to the bar chart and pie chart visualization types instead of over here where the code is more generic.